src/Core/Managed/Net45/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -0,0 +1,498 @@
+                        exceptionData.problemId,
+                        exceptionData.properties,
+                        exceptionData.severityLevel,
+                        exceptionData.ver
+                    }
+                }.GetType());
+                this.telemetryHandlers.Add(typeof(ExceptionTelemetry), new TelemetryHandler(ExceptionTelemetry.TelemetryName, writeMethod, keywords, eventSourceOptions));
+
+                // PerformanceCounter
+                eventSourceOptions = Activator.CreateInstance(eventSourceOptionsType);
+                keywords = Keywords.PerformanceCounters;
+                eventSourceOptionsKeywordsProperty.SetValue(eventSourceOptions, keywords);
+                var perfData = new PerformanceCounterData();
+                writeMethod = writeGenericMethod.MakeGenericMethod(new
+                {
+                    PartA_iKey = partA_iKey_Value,
+                    PartA_Tags = partA_Tags_Value,
+                    PartB_PerformanceCounterData = new { perfData.categoryName, perfData.count, perfData.counterName, perfData.instanceName, perfData.kind, perfData.max, perfData.min, perfData.properties, perfData.stdDev, perfData.value, perfData.ver }
+                }.GetType());
+                this.telemetryHandlers.Add(typeof(PerformanceCounterTelemetry), new TelemetryHandler(PerformanceCounterTelemetry.TelemetryName, writeMethod, keywords, eventSourceOptions));
+
+                // PageView
+                eventSourceOptions = Activator.CreateInstance(eventSourceOptionsType);
+                keywords = Keywords.PageViews;
+                eventSourceOptionsKeywordsProperty.SetValue(eventSourceOptions, keywords);
+                var pageViewData = new PageViewData();
+                writeMethod = writeGenericMethod.MakeGenericMethod(new
+                {
+                    PartA_iKey = partA_iKey_Value,
+                    PartA_Tags = partA_Tags_Value,
+                    PartB_PageViewPerfData = new { pageViewData.duration, pageViewData.measurements, pageViewData.name, pageViewData.properties, pageViewData.url, pageViewData.ver }
+                }.GetType());
+                this.telemetryHandlers.Add(typeof(PageViewTelemetry), new TelemetryHandler(PageViewTelemetry.TelemetryName, writeMethod, keywords, eventSourceOptions));
+
+                // SessionState
+                eventSourceOptions = Activator.CreateInstance(eventSourceOptionsType);
+                keywords = Keywords.SessionState;
+                eventSourceOptionsKeywordsProperty.SetValue(eventSourceOptions, keywords);
+                var sessionStateData = new SessionStateData();
+                writeMethod = writeGenericMethod.MakeGenericMethod(new
+                {
+                    PartA_iKey = partA_iKey_Value,
+                    PartA_Tags = partA_Tags_Value,
+                    PartB_SessionStateData = new { sessionStateData.state, sessionStateData.ver }
```

Comment:
Unfortunately we have to use anonymous type as 4.5 project doesn't support EventDataAttribute. It's a requirement to emit ETW RichPayload event. I will add documents.